libraries/Microsoft.Bot.Connector/Authentication/MicrosoftGovernmentAppCredentials.cs

```diff
@@ -62,5 +65,34 @@ public override string OAuthEndpoint
         {
             get { return GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl; }
         }
+
+        protected override Lazy<IAuthenticator> BuildIAuthenticator()
+        {
+            return new Lazy<IAuthenticator>(
+                () =>
+                {
+                    var credential = new ClientCredential(MicrosoftAppId, MicrosoftAppPassword);
+
+                    IAuthenticator Make(string authority, string scope)
+                        => new AdalAuthenticator(
+                            credential,
+                            new OAuthConfiguration() { Authority = authority, Scope = scope },
+                            this.CustomHttpClient,
+                            this.Logger);
+
+                    if (OAuthScope == GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope)
+                    {
+                        var optionV1 = Make(GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl, GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope);
+                        var optionV2 = Make(GovernmentAuthenticationConstants.ToChannelFromBotLoginUrlV2, GovernmentAuthenticationConstants.ToChannelFromBotOAuthScopeV2);
+
+                        return new Authenticators(optionV2, optionV1);
+                    }
+                    else
+                    {
+                        return Make(OAuthEndpoint, OAuthScope);
+                    }
+                },
+                LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
What's this, and are we sure it's right? I have no idea what it does.

Reply:
I'm following the existing style:

https://github.com/microsoft/botbuilder-dotnet/blob/55af9699614e2ff16a662c5e47510be63f165683/libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs#L144